### PR TITLE
Feature/insights widget

### PR DESCRIPTION
**Pull request by DmitriVanGuard**
#### Summary
- `Widget` UI Component;
- `Insights` widget slider;
- Interaction types: swiping (touch), mouse swipes, buttons, slideshow;
- The Widget displays 5 latest insights;
- Click on the title will lead to the corresponding insight page;
- Click on the author will lead to the page with insights filtered by that author.

#### Screenshots
Insights Widget

### CODE REVIEW

File: app/src/components/InsightsWidget/InsightsWidgetItem.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  text,
+  user: { username, id: userId }
+}) => {
+  const insightText = getTagsStrippedText(text)
+  return (
+    <div className='InsightsWidgetItem'>
+      <h2 className='InsightsWidgetItem__title'>
+        <Link to={`/insights/${id}`}>{title}</Link>
+      </h2>
+      <p className='InsightsWidgetItem__text'>{insightText}</p>
```
[the remark this answers is not in the document]
Comment (Partysun) [reply app/src/components/InsightsWidget/InsightsWidgetItem.js]: @DmitriVanGuard

File: app/src/components/InsightsWidget/InsightsWidget.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  autoplaySpeed: 7000,
+  autoplay: true,
+  arrows: false
+}
+
+const parseInsightsWidgetGQLProps = ({ data: { allInsights = [] } }) => ({
+  insights: allInsights.slice(0, 5)
+})
+
+export class InsightsWidget extends Component {
```
[the remark this answers is not in the document]
Comment (DmitriVanGuard) [reply app/src/components/InsightsWidget/InsightsWidget.js]: It's a mistake. Should be a functional component. I will change it 👍

File: app/src/components/InsightsWidget/InsightsWidgetItem.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (DmitriVanGuard) [reply app/src/components/InsightsWidget/InsightsWidgetItem.js]: Right now the text is truncated by CSS styles. Only the first 3 rows, that fit in the widget, are shown.

File: app/src/components/InsightsWidget/InsightsWidget.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  speed: 500,
+  slidesToShow: 1,
+  slidesToScroll: 1,
+  autoplaySpeed: 7000,
+  autoplay: true,
+  arrows: false
+}
+
+const parseInsightsWidgetGQLProps = ({ data: { allInsights = [] } }) => ({
+  insights: allInsights.slice(0, 5)
```
[the remark this answers is not in the document]
Comment (DmitriVanGuard) [reply app/src/components/InsightsWidget/InsightsWidget.js]: Good point!

**Pull request merged**